PhpAmqpLib/Connection/AMQPConnectionConfig.php

```diff
@@ -604,4 +613,21 @@ public function setConnectionName(string $connectionName): void
     {
         $this->connectionName = $connectionName;
     }
+
+    public function getSocketProtocolMode(): ?int
+    {
+        return $this->socketProtocolMode;
+    }
+
+
+    /**
+     * @param int|null $socketProtocolMode Parameter specifies the protocol family to be used by the socket.<br>
+     *    Const AF_INET IPv4 Internet based protocols.<br>
+     *    Const AF_INET6 IPv6 Internet based protocols.<br>
+     *    null Auto select
+     */
+    public function setSocketProtocolMode(?int $socketProtocolMode): void
+    {
+        $this->socketProtocolMode = $socketProtocolMode;
```

Comment:
This allows to set any integer, while valid values are only 2?

Reply:
[PhpAmqpLib/Connection/AMQPConnectionConfig.php](https://github.com/php-amqplib/php-amqplib/pull/1187/files/1a0b55347dc81ea698abfed76e2a251225440ac6#diff-605f66b52165ee190a37b0110e3eb0a05034a6e1d4573becfd467191fd339fdb)